models/default_local_provider.go

```diff
@@ -1221,3 +1252,163 @@ func getSeededComponents(comp string, log logger.Handler) ([]string, []string, e
 	}
 	return names, contents, nil
 }
+
+// Below helper functions are for downloading seed content and can be re used in future as the way of extracting them changes, like endpoint changes or so. That is why, they are encapsulated into general functions
+func downloadContent(comp string, downloadpath string, log logger.Handler) error {
+	switch comp {
+	case "Pattern":
+		walk := walker.NewGit()
+		return walk.Owner("service-mesh-patterns").Repo("service-mesh-patterns").Root("samples/").Branch("master").RegisterFileInterceptor(func(f walker.File) error {
+			path := filepath.Join(downloadpath, f.Name)
+			file, err := os.Create(path)
+			if err != nil {
+				return err
+			}
+			defer file.Close()
+			fmt.Fprintf(file, "%s", f.Content)
+			return nil
+		}).Walk()
+	case "Filter":
+		return getFiltersFromWasmFiltersRepo(downloadpath)
+	case "Application":
+		var path string
+		path = os.Getenv("SEED_APP_PATH")
+		if path == "" {
+			wd, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+			path = filepath.Join(wd, "../install/seed_apps.json")
+		}
+		applicationsAndURLS, err := getSeededAppLocation(path)
+		if err != nil {
+			return err
+		}
+		for app, URLs := range applicationsAndURLS {
+			yamlFile := filepath.Join(downloadpath, app)
+			f, err := os.Create(yamlFile)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			err = downloadYAMLSintoSingleFile(f, URLs)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func downloadApplicationsFromURL(downloadpath string, appname string, url string) error {
+	path := filepath.Join(downloadpath, appname)
+	res, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	fmt.Fprintf(f, "%s", content)
+	return nil
+}
+
+//takes a slice of URL's which each returns a YAML body on get request. Then combines all the yamls into one yaml
+func downloadYAMLSintoSingleFile(f io.Writer, URLs []string) error {
+	for _, url := range URLs {
+		res, err := http.Get(url)
+		if err != nil {
+			return err
+		}
+		content, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(f, "%s\n---\n", string(content))
+	}
+	return nil
+}
+
+func getFiltersFromWasmFiltersRepo(downloadPath string) error {
+	var helper map[string]interface{}
+	res, err := http.Get("https://api.github.com/repos/layer5io/wasm-filters/releases/latest")
```

Comment:
Can we avoid use of GitHub APIs in order to avoid rate limiting?

Reply:
@Revolyssup I fixed the releases in https://github.com/layer5io/wasm-filters/releases, and so you should be successful in switching over to https://github.com/layer5io/wasm-filters/releases/latest now.

Reply:
https://github.com/layer5io/wasm-filters/releases/latest works now.